### PR TITLE
feat: Claim Third Year Anniversary Campaign

### PR DESCRIPTION
**Pull request by ChefMomota**
### Summary
🎂🎁📜

This pull request adds the code and components for the anniversary achievement campaign, which rewards users with points and achievements for being active on PancakeSwap for at least two years. It refactors the `GlobalCheckClaimStatus` component to handle different campaigns in separate files and replaces the V3 airdrop modal with the anniversary achievement modal. It also adds the contract address, ABI, and helper functions for the anniversary achievement contract, which is used by the `useContract` hook to interact with the contract.

> _We celebrate the third year of PancakeSwap_
> _We claim our points and achievements with `useContract`_
> _We interact with the contract using the ABI_
> _We unleash the power of the anniversary campaign_

### Walkthrough
*  Update the anniversary achievement modal to reflect the 3rd year anniversary ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-ea735f18b640a996378d1c2a3c429c32faf66aaaa78a4287dabc9a37fbca7dabL61-R69))
*  Move the code of the `GlobalCheckClaimStatus` component to the `AnniversaryAchievementModal.tsx` file for better modularity ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-db39cfb6726323b2ef55ab35980fd64b186a0e3948ed2e3040bd46528529afe4L1-R12))
*  Implement the logic of checking the claim status for the anniversary achievement campaign using the `canClaim` function of the contract and displaying the modal if the user can claim ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-db39cfb6726323b2ef55ab35980fd64b186a0e3948ed2e3040bd46528529afe4L34-R92), [link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4R51), [link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4R410-R415))
*  Add the code of the anniversary achievement ABI to the `anniversaryAchievement.ts` file to provide the interface for interacting with the contract ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-0a7f72caeeea5a5a1e3c0bf8779ba47782d19a3c2bee1f83b5363bb4d6ca95faR1-R132))
*  Add the code of the anniversary achievement campaign to the `campaigns.ts` file to define the properties of the campaign for the profile page and the modal ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-93c5ec9c349ca48e1af55348f38a9f2d2b9eb74daf96d6fb7590329b36729a37R219-R225))
*  Add the code of the anniversary achievement contract address to the `contracts.ts` file to provide the address for different chain ids ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-6ccab19ffbec70686dc2221f7b79432ece4302c36067c91063d258209c09ec07R256-R259))
*  Add the code of the `getAnniversaryAchievementAddress` function to the `addressHelpers.ts` file to create a helper function that returns the contract address for a given chain id ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-05729caa7bf80ee8f359513f222175a79f3cd3e8aa704d68f0d5c67f8b010a23R151-R154))
*  Export the `getAnniversaryAchievementAddress` function from the `addressHelpers.ts` file and import the anniversary achievement ABI in the `contractHelpers.ts` file ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-dbcbcd450dce97f74ff9b8fff28d36fc457775363d349db2a45e504060c0754aR37), [link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-dbcbcd450dce97f74ff9b8fff28d36fc457775363d349db2a45e504060c0754aR85))
*  Add the code of the `getAnniversaryAchievementContract` function to the `contractHelpers.ts` file to create a helper function that returns the contract instance using the ABI and the address ([link](https://github.com/pancakeswap/pancake-frontend/pull/7790/files?diff=unified&w=0#diff-dbcbcd450dce97f74ff9b8fff28d36fc457775363d349db2a45e504060c0754aR415-R423))

### CODE REVIEW

File: apps/web/src/config/constants/campaigns.ts [hunk 1]
```diff
@@ -215,6 +215,13 @@ const campaigns: Campaign[] = [
     description: 'Celebrate our 2nd birthday with us',
     badge: '2-year.svg',
   },
+  {
+    id: '516020001', // TODO confirm with SC
```
Comment (ChefMomota) [inline apps/web/src/config/constants/campaigns.ts:219]: Change this before merge

File: apps/web/src/config/constants/contracts.ts [hunk 2]
```diff
@@ -253,4 +253,8 @@ export default {
     [ChainId.BSC]: '0xCD5d1935e9bfa4905f9f007C97aB1f1763dC1607',
     [ChainId.BSC_TESTNET]: '0xd2d1DD41700d9132d3286e0FcD8D6E1D8E5052F5',
   },
+  anniversaryAchievement: {
+    [ChainId.BSC]: '0xF839286bD9D14b358496829F3BaB3145C16ad3C1',
```
Comment (ChefMomota) [inline apps/web/src/config/constants/contracts.ts:257]: Change this before merge

File: apps/web/src/config/constants/campaigns.ts (hunk 1 above)
Comment (ChefMomota) [reply apps/web/src/config/constants/campaigns.ts:219]: 516030001

File: apps/web/src/config/abi/anniversaryAchievement.ts
Comment (chefjackson) [inline apps/web/src/config/abi/anniversaryAchievement.ts:1]: I recall we have a similar json file. Can we remove that one?

Comment (ChefMomota) [reply apps/web/src/config/abi/anniversaryAchievement.ts:1]: done

**Review by chefjackson — APPROVED** [review]

**Pull request merged**